src/plugins/posix/uring_queue.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "uring_queue.h"
+#include <liburing.h>
+#include <array>
+#include <vector>
+#include <cstring>
+#include <stdexcept>
+#include "common/status.h"
+#include "common/nixl_log.h"
+
+namespace {
+    static constexpr unsigned int max_posix_ring_size_log = 10;
+    static constexpr unsigned int max_posix_ring_size = 1 << max_posix_ring_size_log;
+}
+
+nixl_status_t UringQueue::init(int entries, const io_uring_params& params, bool read_op) {
+    num_entries = entries;
+    num_completed = 0;
+    num_submitted = 0;
+    is_read = read_op;
+    memset(&uring, 0, sizeof(uring));
+
+    // Start with basic parameters, no special flags
+    io_uring_params local_params = params;
+
+    // Initialize with basic setup
+    int ret = io_uring_queue_init_params(entries, &uring, &local_params);
+    if (ret < 0) {
+        NIXL_ERROR << "Failed to initialize io_uring instance: " << strerror(-ret);
+        return NIXL_ERR_BACKEND;
+    }
+
+    // Log the features supported by this io_uring instance
+    NIXL_INFO << "io_uring features:"
+              << " SQPOLL=" << ((local_params.features & IORING_FEAT_SQPOLL_NONFIXED) ? "yes" : "no")
+              << " IOPOLL=" << ((local_params.features & IORING_FEAT_FAST_POLL) ? "yes" : "no");
+
+    return NIXL_SUCCESS;
+}
+
+UringQueue::UringQueue(int num_entries, const io_uring_params& params, bool is_read) {
+    nixl_status_t status = init(num_entries, params, is_read);
+    if (status != NIXL_SUCCESS) {
+        throw std::runtime_error("Failed to initialize UringQueue");
+    }
+}
+
+UringQueue::~UringQueue() {
+    io_uring_queue_exit(&uring);
+}
+
+nixl_status_t UringQueue::submit() {
+    int ret = io_uring_submit(&uring);
+    if (ret < 0) {
```

Comment:
if 0 < ret < num_entries, then some submissions failed, and the user won't know
```suggestion
    if (ret != num_etnries) {
```